src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
 
         return output_result
 
-    def validate_experiment(self, path: Path) -> Tuple[bool, str]:
+    def validate_experiment(self, path: Path) -> ErrorDictType:
         # TODO: Add any other validation which should be done
         """
         Validates the experiment by checking:
         - if the structure is correct and consists of an experiment.json
         - (For local run) experiment directory contains an input directory with the correct files
-        - (For local run) check the python & yaml files for correct syntax (possible?)
         - content of experiment.json is valid JSON
         - asset in the experiment.json validated against a schema validator.
+        - if the network used in experiment.json is existing
+        - if the nodes and channels used in experiment.json are correct and valid for that network
+
+        Args:
+            path: The location of the experiment
+
+        Returns:
+            ErrorDictType: Dict[str, List[str]]: Dictionary containing error, warning and info messages of the
+            validations that failed
         """
 
         experiment_json = path / 'experiment.json'
+        local = True
+        error_dict: ErrorDictType = {"error": [], "warning": [], "info": []}
+
+        # Check if experiment.json exists
         if not experiment_json.is_file():
-            return False, 'File experiment.json not found in the current working directory'
+            error_dict['error'].append(f'File not found: {experiment_json} does not exist')
+
+        # Check if experiment is local or remote
+        if experiment_json.is_file():
+            is_valid, message = validate_json_file(experiment_json)
+            if is_valid:
+                experiment = read_json_file(experiment_json)
+                if not experiment['meta']['backend']['location'] == "local":
```

Comment:
You didn't solve it, because now the same comment stands for experiment['meta']['backend']. Who says meta is a key and has a backend. What I meant is do first the schema validation on this file first, then you know these fields are in and valid to use. Now you do the schema validation in the last step.